packages/twenty-front/src/modules/object-record/record-table/record-table-cell/components/RecordTableCellFocusedPortal.tsx

```diff
@@ -0,0 +1,149 @@
+import { RecordTableCellPortalWrapper } from '@/object-record/record-table/record-table-cell/components/RecordTableCellPortalWrapper';
+import { useRecoilComponentValue } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValue';
+
+import { hasRecordGroupsComponentSelector } from '@/object-record/record-group/states/selectors/hasRecordGroupsComponentSelector';
+import { TABLE_Z_INDEX } from '@/object-record/record-table/constants/TableZIndex';
+import { RecordTableCellFocusedPortalContent } from '@/object-record/record-table/record-table-cell/components/RecordTableCellFocusedPortalContent';
+import { RecordTableCellPortalRootContainer } from '@/object-record/record-table/record-table-cell/components/RecordTableCellPortalRootContainer';
+import { isRecordTableScrolledHorizontallyComponentState } from '@/object-record/record-table/states/isRecordTableScrolledHorizontallyComponentState';
+import { isRecordTableScrolledVerticallyComponentState } from '@/object-record/record-table/states/isRecordTableScrolledVerticallyComponentState';
+import { recordTableFocusPositionComponentState } from '@/object-record/record-table/states/recordTableFocusPositionComponentState';
+import { recordTableHoverPositionComponentState } from '@/object-record/record-table/states/recordTableHoverPositionComponentState';
+import { isDefined } from 'twenty-shared/utils';
+
+export const RecordTableCellFocusedPortal = () => {
+  const focusPosition = useRecoilComponentValue(
+    recordTableFocusPositionComponentState,
+  );
+
+  const isRecordTableScrolledVertically = useRecoilComponentValue(
+    isRecordTableScrolledVerticallyComponentState,
+  );
+
+  const isRecordTableScrolledHorizontally = useRecoilComponentValue(
+    isRecordTableScrolledHorizontallyComponentState,
+  );
+
+  const hasRecordGroups = useRecoilComponentValue(
+    hasRecordGroupsComponentSelector,
+  );
+
+  const hoverPosition = useRecoilComponentValue(
+    recordTableHoverPositionComponentState,
+  );
+
+  const isUnderHoveredPortal =
+    isDefined(hoverPosition) &&
+    isDefined(focusPosition) &&
+    hoverPosition.column === focusPosition.column &&
+    hoverPosition.row === focusPosition.row;
+
+  if (!isDefined(focusPosition) || isUnderHoveredPortal) {
+    return null;
+  }
+
+  const isOnFirstScrollableColumn = focusPosition.column === 1;
+  const isOnLabelIdentifierStickyColumn = focusPosition.column === 0;
+
+  const zIndexForHoveredPortalOnFirstScrollableColumnWithoutGroups =
```

Comment:
this looks very hard to maintain, we should make functions and add tests on top of it